pkg/plugin/processor/builtin/impl/openai-textgen/textgen.go

```diff
@@ -0,0 +1,233 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package textgen
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/sashabaranov/go-openai"
+)
+
+//go:generate paramgen -output=paramgen_proc.go ProcessorConfig
+
+type Processor struct {
```

Comment:
```suggestion
type textgenProcessor struct {
```
It aligns with the naming convention used in other built-in processors.